src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,124 @@
+import {
+  useState,
+  useEffect,
+  useMemo,
+} from 'react';
+import { useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
+import { Loader, PeopleTable, PeopleFilters } from '../components';
+import {
+  ERRORS,
+  getPeople,
+  getPeopleSorted,
+  getPeopleFiltered,
+  getPeoplePrepared,
+} from '../utils';
+import { Person, SearchParameters } from '../types';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isDataFetched, setIsDataFetched] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [searchParams] = useSearchParams();
+  const query = searchParams.get(SearchParameters.Query) || '';
+  const filterBySex = searchParams.get(SearchParameters.Sex) || '';
+  const centuries = searchParams.getAll(SearchParameters.Centuries) || [];
+  const sort = searchParams.get(SearchParameters.Sort) || '';
+  const order = searchParams.get(SearchParameters.Order) || '';
+
+  useEffect(() => {
+    setIsLoading(true);
+    const fetchPeople = async () => {
+      try {
+        const currentPeople = await getPeople();
+
+        if (!currentPeople.length) {
+          setErrorMessage(ERRORS.NO_PEOPLE_ERROR);
+        }
+
+        setPeople(getPeoplePrepared(currentPeople));
+        setIsDataFetched(true);
+      } catch (error) {
+        setErrorMessage(ERRORS.DOWNLOAD_ERROR);
+        setIsDataFetched(false);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchPeople();
+  }, []);
+
+  const filteredPeople = useMemo(() => {
+    return getPeopleFiltered(filterBySex, query, centuries, people);
+  }, [filterBySex, query, centuries, people]);
+
+  const visiblePeople = useMemo(() => {
+    return getPeopleSorted(sort, order, filteredPeople);
+  }, [sort, order, filteredPeople]);
+
+  const noPeopleOnSearchErrorCondition = !visiblePeople.length
+  && isDataFetched && !errorMessage;
+
+  useEffect(() => {
+    if (noPeopleOnSearchErrorCondition) {
+      setErrorMessage(ERRORS.NO_PEOPLE_ON_SEARCH_ERROR);
+    } else if (errorMessage !== ERRORS.NO_PEOPLE_ERROR) {
+      setErrorMessage('');
+    }
+  }, [visiblePeople.length, isDataFetched]);
+
+  const isTableVisible = !errorMessage && !!visiblePeople.length;
+
+  const isFilterPanelVisible = isTableVisible
+  || errorMessage === ERRORS.NO_PEOPLE_ON_SEARCH_ERROR;
```

Comment:
It is not highly important but try to follow the default order of things in `react` component:

1. hooks
2. variables
2. effects
3. methods
4. render

Reply:
I thought that the last error is only detected in the second useEffect, that's why this order.